### PR TITLE
Initial IPv6 support

### PR DESCRIPTION
**Pull request by BenTheElder**
@aojea's PR (#348) is really close, seeing if we can finish it up before 0.4 :-)

### CODE REVIEW

**Comment by BenTheElder** [comment]
the last commit fixed the job (still failing, but only because some of the conformance test ipv6 issues that are known and don't appear to be kind specific)

**Comment by BenTheElder** [comment]
locally tested with single node :wink: 
this is awesome. no major remaining issues. thank you so much @aojea :tada:

**Comment by BenTheElder** [comment]
(self lgtm because my additional commit is a trivial fix, and I already approve all of the remaining code :-))

**Comment by aojea** [comment]
Omg I'm almost crying with the emotion

**Comment by neolit123** [comment]
nice work, guys. :tada:

**Comment by BenTheElder** [comment]
Let's go get a CI job up!

File: site/content/docs/user/quick-start.md [hunk 1]
```diff
@@ -347,3 +363,4 @@ kind, the Kubernetes cluster itself, etc.
 [Kubernetes imagePullPolicy]: https://kubernetes.io/docs/concepts/containers/images/#updating-images
 [Private Registries]: /docs/user/private-registries
 [customize control plane with kubeadm]: https://kubernetes.io/docs/setup/independent/control-plane-flags/
+[docker enable ipv6]: https://docs.docker.com/config/daemon/ipv6/
```
Comment (PatrickLang) [inline site/content/docs/user/quick-start.md]: the instructions on that page don't set `fixed-cidr-v6` so the daemon fails to start

Comment (BenTheElder) [reply site/content/docs/user/quick-start.md]: I remember running into this now... shall we send them a PR? 🙃 
https://github.com/docker/docker.github.io/edit/master/config/daemon/ipv6.md

Comment (aojea) [reply site/content/docs/user/quick-start.md]: They have it documented here https://docs.docker.com/v17.09/engine/userguide/networking/default_network/ipv6/ , don´t know why they didn´t explain it in that link

**Comment by BenTheElder** [comment]
Should we switch links then?

**Pull request merged**